### ISSUE TITLE
Edit EP Claim Labels | Add UpdateBenefitClaim endpoint to ruby-bgs

### ISSUE THREAD

**Issue by leikkisa** [department-of-veterans-affairs/caseflow#14888]
## Description
We'd like to implement the BGS service for updating claims, in order to use it for Edit EP claim labels.  This should allow us to update the EP type code without losing any work that's happened so far on the claim.  Updating the claim label should automatically re-route the claim in VBMS.

## Acceptance criteria
- [ ] Implement `updateBenefitClaim` in **BenefitClaimWebService** in ruby-bgs
  - [ ] Fields to include in the request:
    - [ ] benefitClaimUpdateInput:
      - [ ] fileNumber (required)
      - [ ] payeeCode
      - [ ] dateofClaim
      - [ ] benefitClaimType
      - [ ] oldEndProductCode
      - [ ] newEndProductLabel
      - [ ] oldDateOfClaim
  - [ ] Return the benefitClaimRecord
- [ ] Test ruby-bgs PR in UAT by updating an EP's code

## Open questions
- Which fields are required?
- Double check with Colin if this request actually re-routes the claim

## Out of scope
- Implementing in Caseflow or testing Caseflow UAT

## Technical notes
Documentation for BenefitClaimWebService can be found here:

https://github.com/department-of-veterans-affairs/appeals-team/tree/master/Project%20Folders/Caseflow%20Projects/Intake/BGS-VBMS/BGS-API-Docs

**Comment by leikkisa** [comment]
Here is an example SOAP doc from Nathan Jolly:
```xml
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ser="http://services.share.benefits.vba.va.gov/">
   ${#Project#header}
   <soapenv:Body>
      <ser:updateBenefitClaim>
         <!--Optional:-->
         <benefitClaimUpdateInput>
            <fileNumber>792647867</fileNumber>
            <payeeCode>00</payeeCode>
            <dateOfClaim>09/05/2018</dateOfClaim>
            <disposition>M</disposition>        
            <sectionUnitNo>1911</sectionUnitNo>
            <folderWithClaim>N</folderWithClaim>
            <gulfWarRegistryPermit>Y</gulfWarRegistryPermit>
            <benefitClaimType>1</benefitClaimType>
            <oldEndProductCode>810</oldEndProductCode>
            <newEndProductLabel>810LROSU</newEndProductLabel>
            <oldDateOfClaim>09/05/2018</oldDateOfClaim>
         </benefitClaimUpdateInput>
      </ser:updateBenefitClaim>
   </soapenv:Body>
</soapenv:Envelope>
```

**Comment by leikkisa** [comment]
Adding a comment from Linda that got added after the last BGS meeting, regarding date format:

> One thing I can see is it doesn't have a full date format, another wonderful feature of 12C was that we need that format now.

**Comment by leikkisa** [comment]
We are asking some folks about the values to use for disposition, sectionUnitNo, and folderWithClaim, but the default assumption is to use the same values that we hardcode for these fields when establishing claims in VBMS.  That unblocks this PR for now, however let's not merge it until after Thursday in case we get some comments on our inquiry or hear anything during the Thursday integrations meeting.
